### PR TITLE
salt-api return consistent data at local, local_async client

### PR DESCRIPTION
**Pull request by GwiYeong**
add feature for `salt-api return consistent data at local, local_async client`
refer PR #53503 

~~writing version warning until Magnesium~~ 
write warning to date `20210101`

### Tests written?
Yes but commented

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by GwiYeong** [comment]
@waynew 
Hi, i rewrite code for the PR #53503
Please review for this. 

thanks !

**Review by twangboy — CHANGES_REQUESTED** [review]
Looks like this is failing the linter

File: salt/netapi/rest_tornado/saltnado.py [hunk 1]
```diff
@@ -1118,6 +1118,20 @@ def _disbatch_local_async(self, chunk):
...
+
+            import salt.utils.versions
+            salt.utils.versions.warn_until(
+                'Magnesium',
+                'The empty data returning when jid not exist pub_data will be deprecated.'
+                'Please ensure the message return like '
+                '"No minions matched the target. No command was sent, no jid was assigned" if jid not exist after published.'
+                'this change is for consistency of salt-api return (See PR #53503)'
+            )
+
```
Comment (twangboy) [inline salt/netapi/rest_tornado/saltnado.py]: This is failing the linter... too many blank lines

File: tests/integration/netapi/rest_tornado/test_app.py [hunk 2]
```diff
@@ -292,6 +292,9 @@ def test_simple_local_async_post_no_tgt(self):
                               )
         response_obj = salt.utils.json.loads(response.body)
         self.assertEqual(response_obj['return'], [{}])
+        # TODO uncomment after deprecated (See PR #53503)
+        # self.assertEqual(response_obj['return'], ["No minions matched the target. No command was sent, no jid was assigned."])
+
```
Comment (twangboy) [inline tests/integration/netapi/rest_tornado/test_app.py]: Also failing the linter... too many blank lines

File: salt/netapi/rest_tornado/saltnado.py (hunk 1 above)
Comment (GwiYeong) [reply salt/netapi/rest_tornado/saltnado.py]: i fixed it. thanks

File: tests/integration/netapi/rest_tornado/test_app.py (hunk 2 above)
Comment (GwiYeong) [reply tests/integration/netapi/rest_tornado/test_app.py]: i fixed it. thanks

**Comment by waynew** [comment]
Hi @GwiYeong, thanks for the PR, and sorry for the delay - I just saw this update.

Could you rebase this on master? We've changed to a master development branch.

Thanks!

**Comment by GwiYeong** [comment]
@waynew 
Hi. rebase from develop to master just completed.

thanks.

File: salt/netapi/rest_tornado/saltnado.py [hunk 3]
```diff
@@ -1113,6 +1113,19 @@ def _disbatch_local_async(self, chunk):
         f_call = self._format_call_run_job_async(chunk)
         # fire a job off
         pub_data = yield self.saltclients['local_async'](*f_call.get('args', ()), **f_call.get('kwargs', {}))
+        if 'jid' not in pub_data:
+            # TODO uncomment after deprecated
+            # please check test case for this tests.integration.netapi.rest_tornado.test_app:test_simple_local_async_post_no_tgt
+            # raise tornado.gen.Return('No minions matched the target. No command was sent, no jid was assigned.')
+
+            import salt.utils.versions
+            salt.utils.versions.warn_until(
```
Comment (waynew) [inline salt/netapi/rest_tornado/saltnado.py]: With our new process, we're making it a flat year for deprecations.

Could you change this to `warn_until_date('20210101', ...`?

**Review by waynew — CHANGES_REQUESTED** [review]

File: salt/netapi/rest_tornado/saltnado.py (hunk 3 above)
Comment (GwiYeong) [reply salt/netapi/rest_tornado/saltnado.py]: sorry for late response.
changed warning with date.

**Comment by dwoz** [comment]
Closing this due to inactivity.  Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**